.github/workflows/ci.yaml

```diff
@@ -17,17 +17,22 @@ jobs:
     steps:
       - name: Checkout code
         uses: actions/checkout@v4
+      - name: Cache Go Modules
+        uses: actions/cache@v4
         with:
-          fetch-depth: 1
+          path: |
+            ~/.cache/go-build
+            ~/go/pkg/mod
+          key: ${{ runner.os }}-gomod-ci-${{ hashFiles('gradle.properties', 'gradle/libs.versions.toml') }}
+          restore-keys:
+            ${{ runner.os }}-gomod-ci-
```

Comment:
For the ci/release workflows, we now cache go modules and build for building the license header binary. We no longer need to build the buf binary as we pull it from the Maven artifact.